src/middlewared/middlewared/validators.py

```diff
@@ -87,7 +87,7 @@ def __init__(self, pattern, flags=0, explanation=None):
         self.regex = re.compile(pattern, flags)
 
     def __call__(self, value):
-        if value is not None and not self.regex.match(value):
```

Comment:
Empty string should still be validated by the regex and this change can cause regressions for all the `Match` validator usages. We should add a flag to the validator for `empty` defaulting to `false` and overriding to `true` for cases like these where it's okay to have empty string i think.